### DISCUSSION TITLE
UI integration and credit templates

### DISCUSSION THREAD

**Discussion by ghuDaYuYu** [ghDaYuYu/foo_discogs#93] [Ideas]
Although foo_discogger v1.0.x will still be fixed and updated from time to time, it certainly seems quite stable now,
meaning that with the exception of a few pending issues, there is little left to do.
Thanks for supporting me this long!

I am posting v1.1.0.alpha to provide some insight into medium term development goals, mainly UI integration and credit templates. Hopefully bug/crashes won´t be so elusive...

![1 1 0 Alpha](https://user-images.githubusercontent.com/62167612/179420328-0fc71b31-1bf6-4697-8cf9-542649e989e4.JPG)

Disclaimer!
This is quite unstable, specially in multitask scenarios...
Run it on test media/portable fb2k installation to protect your media library.

[zipped 1.1.0Alpha](https://github.com/ghDaYuYu/foo_discogs/files/9128348/zipped.1.1.0Alpha.zip)

**Comment by ghuDaYuYu** [comment]
1.1.0.1Alpha:

- Fixed crash on UI element right-click 'Write tags...' on some scenarios
- Fixed extra spaces on credit disk.track formats
- SDK 2022-08-10 (includes UI element vertical scrollbar fix)

[zipped v1.1.0.1Alpha](https://github.com/ghDaYuYu/foo_discogs/files/9334081/zipped.v1.1.0.1Alpha.zip)

**Comment by ghuDaYuYu** [comment]
1.1.0.2 Alpha

- x64 Release
- Dark mode (excl. icons, artist list and artwork panel)
- Fixed crash right clicking on empty selections

![1 1 0 2](https://user-images.githubusercontent.com/62167612/188289302-f087d30c-c485-419b-8906-a510efc5eaa3.png)

[foo_discogger v1.1.0.2 Alpha](https://github.com/ghDaYuYu/foo_discogs/files/9483475/foo_discogger.v1.1.0.2.Alpha.zip)

[foo_discogger.x64 v1.1.0.2 Alpha](https://github.com/ghDaYuYu/foo_discogs/files/9483478/foo_discogger.x64.v1.1.0.2.Alpha.zip)

**Comment by Lovecraft-github** [comment]
Not related at all.....

My god, Vampyros Lesbos, the Jesús Franco cult movie.

**Comment by ghuDaYuYu** [comment]
1.1.17.4 Alpha

- fb2k SDK update
- include code updates from releases v1.0.11 to v1.0.17.4
- v1.1.x: fix crash highlighting credit list items

[foo_discogger 1.1.17.4 Alpha](https://github.com/ghDaYuYu/foo_discogs/files/11552106/foo_discogger.1.1.17.4.Alpha.zip)

**Comment by ghuDaYuYu** [comment]
v1.1.17.4 Alpha 2

- fix library view selection crash
- fix unlisted credit exceptions (now 'Other [Unlisted credit]')

Known issues:
- multiple UI instances not supported

[foo_discogger v1.1.17.4 Alpha 2](https://github.com/ghDaYuYu/foo_discogger/files/11713291/foo_discogger.v1.1.17.4.Alpha.2.zip)

**Comment by ghuDaYuYu** [comment]
v1.1.17.4 Alpha 3

Fixes:
- 'in tracks' template filter UI results (single track selections only)
- UI credit double click crash (now copies credit row to clipboard)
- race condition lock (library view selections)

Whats new:
- TMPL_OTHER and TMPL_FEATURING_&_REMIX default templates 

Known issues:
- multiple UI instances not supported

[foo_discogger v1.1.17.4 Alpha 3](https://github.com/ghDaYuYu/foo_discogger/files/11746577/foo_discogger.v1.1.17.4.Alpha.3.zip)

**Comment by ghuDaYuYu** [comment]
v1.1.17.4 Alpha 4

Fixed:

- credit parse exception (extra trailing comma in tracks: 2,6,7,9,)
- additional task cancellation checks (releases with many tracks/multiple discs)
- release credit filter (all, in release, in tracks)

New:

- per track credits tags (%TRACK_CAT_CREDITS...%)
- new view: group credits by track + context menu options

![TrackCreds](https://github.com/ghDaYuYu/foo_discogger/assets/62167612/ce444f7a-f232-4502-a621-bdcb4781daf4)


[foo_discogger v1.1.17.4 Alpha 4](https://github.com/ghDaYuYu/foo_discogger/files/11780730/foo_discogger.v1.1.17.4.Alpha.4.zip)

**Comment by ghuDaYuYu** [comment]
v1.1.18 Alpha 1

- Includes public release [v1.0.18](https://github.com/ghDaYuYu/foo_discogger/releases/tag/v1.0.18) updates.
- Context menu revision, including new option 'Delete release credit cache'.


[foo_discogger v1.1.18 Alpha 1](https://github.com/ghDaYuYu/foo_discogger/files/11971796/foo_discogger.zip)

**Comment by ghuDaYuYu** [comment]
v1.1.20 alpha 1

[foo_discogger v1.1.20 alpha 1](https://github.com/ghDaYuYu/foo_discogger/files/12566123/foo_discogger.v1.1.20.alpha.1.zip)

Change log:

- CUI support.
- Pop-up panel support (Main menu > View > Discogger)

Limitations:
- Single UI instance support.

Getting started (Discogger UI):
- Add Discogger UI to the fb2k layout.
- Clear memory cache.
- Enable read/write disc cache.
- Open the Tag Mapping dialog, add TMPL_ALL_PER_RELEASE and TMPL_TRACK_ALL_PER_TRACK from the 'Credits...' drop-down menu.
- Alternatively add them manually: TMPL_ALL_PER_RELEASE (%RELEASE_CAT_CREDITS_GXP_0_RELEASE%) and TMPL_TRACK_ALL_PER_TRACK (%TRACK_CAT_CREDITS_GXP_0%).

- Select any album from the Library view and write/update DISCOGS_ARTIST_ID, DISCOGS_MASTER_RELEASE_ID and DISCOGS_RELEASE_ID metadata.
- Repeat the previous step with another album.
- Album selections should now display release and track credits in the UI panel.

**Comment by ghuDaYuYu** [comment]
v1.1.20 alpha 2


[foo_discogger v1.1.20 alpha 2](https://github.com/ghDaYuYu/foo_discogger/files/12566667/foo_discogger.v1.1.20.alpha.2.zip)

Change log:
- Removed double-click default action in 'Discogger release credits: Tracks' view.  To run default action press 'Enter' or use Shift or Control modifiers to Send to current playlist, Add to current playlist and Send to new playlist.

Edit: fixed SDK target version

**Comment by ghuDaYuYu** [comment]
(v1.1.20 alpha 2 re-uploaded)

**Comment by ghuDaYuYu** [comment]
[foo_discogger v1.1.20 alpha 3](https://github.com/ghDaYuYu/foo_discogger/files/13060897/foo_discogger.v1.1.20.alpha.3.zip)

Change log:
- Fix artwork manager crash

**Comment by ghuDaYuYu** [comment]
[foo_discogger_v1.1.22 alpha](https://github.com/user-attachments/files/30862874/foo_discogger_v1.1.22.alpha.zip)


Change log:
- Include public release v1.0.22 source updates.

**Reply by Lovecraft-github** [reply]
Hi @ghuDaYuYu 
The credit window doesn't obey the interface colors as the others windows
<img width="1920" height="1028" alt="Captura de pantalla_20260809_142424-1" src="https://github.com/user-attachments/assets/db635111-1a83-449a-95d9-4f1418d88990" />
I'm running it on Kubuntu 26.04 with wine-11.14 (Staging)

**Reply by Lovecraft-github** [reply]
Update:
It's only happening with the default inicial size window, after resizing it works fine.
<img width="900" height="573" alt="Captura de pantalla_20260809_143859" src="https://github.com/user-attachments/assets/429af561-a9fd-463d-ab24-1f7b5845e434" />

**Reply by Lovecraft-github** [reply]
I got a crash trying to tag The Doors - The Doors
zip with dmp and txt
[failure_00000003.zip](https://github.com/user-attachments/files/30874741/failure_00000003.zip)

**Comment by ghuDaYuYu** [comment]
The credit window was patched and this release should be more stable launching UI tasks.

[foo_discogger_v1.1.22 alpha 2.zip](https://github.com/user-attachments/files/30882102/foo_discogger_v1.1.22.alpha.2.zip)

Thanks for testing and the crash report.
I am not sure if this fixed that specific crash, please send dmp and txt again if it didn´t.

**Reply by Lovecraft-github** [reply]
Still crashing
[failure_00000004.zip](https://github.com/user-attachments/files/30891199/failure_00000004.zip)

**Reply by ghuDaYuYu** [reply]
Nice catch!

[foo_discogger_v1.1.22 alpha 3.zip](https://github.com/user-attachments/files/30906425/foo_discogger_v1.1.22.alpha.3.zip)

Change log:
- Remove all text within parentheses from track locations.

(edit: link fixed)

**Reply by Lovecraft-github** [reply]
I can't download alpha 3.404 page
Edit:
Only need to add ".zip" to link

No crash, load the releases normally

Thanks

**Comment by ghuDaYuYu** [comment]
[foo_discogger_v1.1.22.3 alpha 4.zip](https://github.com/user-attachments/files/30940485/foo_discogger_v1.1.22.3.alpha.4.zip)

- Sources updated from v1.0.22.3

And here is a brief test case to compare track credits to credits on the Discogs website:
test release: [Various #36391783](https://www.discogs.com/release/36391783-Various-Steppin-Out-The-Roots-Of-Garage-Rock-1963-1965?redirected=true)
<img width="652" height="269" alt="Various #36391783" src="https://github.com/user-attachments/assets/32800938-c20d-4544-88e1-72ef39c5c8b1" />

Tag mapping: TMPL_TRACK_ALL_PER_TRACK, value: %TRACK_CAT_CREDITS_GXC_0%, action: update.
(per credit)
<img width="549" height="417" alt="TMPL_TRACK_ALL_PER_TRACK %TRACK_CAT_CREDITS_GXC_0%" src="https://github.com/user-attachments/assets/97776115-fa2a-4c1b-b40e-f67bcee22e45" />
Tag mapping: TMPL_TRACK_ALL_PER_TRACK, value: %TRACK_CAT_CREDITS_GXP_0%, action: update.
(per artist)
<img width="429" height="379" alt="TMPL_TRACK_ALL_PER_TRACK %TRACK_CAT_CREDITS_GXP_0%" src="https://github.com/user-attachments/assets/fbaa592b-5f0e-49b7-8a36-09a3af8af61d" />

Still a long way to go... 
Among others, here are some known issues of interest:
- Tag mappings and Credits mappings are not synchronized.
- 'TMPL_' prefix is required to display UI credits.
- Album/track selection and mapping refresh issues.
